server/schedulers/utils.go

```diff
@@ -113,3 +114,138 @@ const (
 func newTaintCache() *cache.TTLUint64 {
 	return cache.NewIDTTL(taintCacheGCInterval, taintCacheTTL)
 }
+
+// ScorePair stores storeID and record of a store.
+type ScorePair struct {
+	storeID uint64
+	score   float64
+}
+
+// NewScorePair returns a ScorePair.
+func NewScorePair(storeID uint64, score float64) *ScorePair {
+	return &ScorePair{
+		storeID: storeID,
+		score:   score,
+	}
+}
+
+// GetStoreID returns the storeID.
+func (s *ScorePair) GetStoreID() uint64 {
+	return s.storeID
+}
+
+// GetScore returns the score.
+func (s *ScorePair) GetScore() float64 {
+	return s.score
+}
+
+// SetScore sets the score.
+func (s *ScorePair) SetScore(score float64) {
+	s.score = score
+}
+
+// ScorePairs is a named type for []*ScorePair with helper methods.
+type ScorePairs []*ScorePair
+
+// Len returns length of slice.
+func (s ScorePairs) Len() int { return len(s) }
+
+// Less returns if one number is less than another.
+func (s ScorePairs) Less(i, j int) bool { return s[i].score < s[j].score }
+
+// Swap switches out two numbers in slice.
+func (s ScorePairs) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+// Mean returns the mean of the data.
+func (s ScorePairs) Mean() float64 {
+	if s.Len() == 0 {
+		return 0
+	}
+
+	var sum float64
+	for _, pair := range s {
+		sum += pair.score
+	}
+
+	return sum / float64(s.Len())
+}
+
+// GetMin returns the min of the data.
+func (s ScorePairs) GetMin() *ScorePair {
```

Comment:
This function is meaningless and confusing. It requires the slice is sorted, but `s[0]` directly is enough for a sorted slice.